cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -91,51 +193,601 @@ public void schedule(Id.Program program, SchedulableProgramType programType, Ite
+                } catch (Throwable t) {
+                  LOG.error("Could not poll stream {}", streamId.getName(), t);
+                }
+              }
+            }, Constants.MetricsCollector.DEFAULT_FREQUENCY_SECONDS, TimeUnit.SECONDS);
+          }
+        }
+      }
+    }
+
+    /**
+     * Add a new scheduling task based on the data received by the stream referenced by {@code this} object.
+     */
+    public void createScheduleTask(Id.Program programId, SchedulableProgramType programType,
+                                   StreamSizeSchedule streamSizeSchedule, boolean active,
+                                   long baseRunSize, long baseRunTs, boolean persist) throws SchedulerException {
+      final StreamSizeScheduleTask newTask = new StreamSizeScheduleTask(programId, programType, streamSizeSchedule);
+      synchronized (this) {
+        StreamSizeScheduleTask previous =
+          scheduleTasks.putIfAbsent(AbstractSchedulerService.scheduleIdFor(programId, programType,
+                                                                           streamSizeSchedule.getName()),
+                                    newTask);
+        if (previous != null) {
+          // We cannot replace an existing schedule - that functionality is not wanted - yet
+          throw new SchedulerException("Tried to overwrite schedule " + streamSizeSchedule.getName());
+        }
+
+        if (active) {
+          activeTasks.incrementAndGet();
+        }
+      }
+
+      StreamSize streamSize;
+      try {
+        streamSize = pollOnce();
+      } catch (IOException e) {
+        // Polling should not fail - the whole logic of stream size schedules relies on it
+        throw new SchedulerException("Polling could not be performed on stream " + streamSizeSchedule.getStreamName(),
+                                     e);
+      }
+
+      // Initialize the schedule task
+      if (baseRunSize == -1 && baseRunTs == -1) {
+        newTask.startSchedule(streamSize.getSize(), streamSize.getTimestamp(), active, persist);
+      } else {
+        newTask.startSchedule(baseRunSize, baseRunTs, active, persist);
+      }
+
+      sendPollingInfoToActiveTasks(new StreamSizeNotification(streamSize.getTimestamp(), streamSize.getSize()));
+    }
+
+    /**
+     * Suspend a scheduling task that is based on the data received by the stream referenced by {@code this} object.
+     */
+    public synchronized void suspendScheduleTask(Id.Program programId, SchedulableProgramType programType,
+                                                 String scheduleName) throws ScheduleNotFoundException {
+      String scheduleId = AbstractSchedulerService.scheduleIdFor(programId, programType, scheduleName);
+      StreamSizeScheduleTask task = scheduleTasks.get(scheduleId);
+      if (task == null) {
+        throw new ScheduleNotFoundException(scheduleName);
+      }
+      if (task.suspend()) {
```

Comment:
what is the contract for this method? If I try to suspend a schedule that is already suspended, is that an error, or is it silently ignored? Please make sure it is consistent with the time schedule and with the documentation (javadocs and dev guide).